Model/Indexer/Product.php

```diff
@@ -2,119 +2,34 @@
-        $areParentsLoaded = false;
-
-        foreach ($storeIds as $storeId) {
-            if ($this->dataHelper->isIndexingEnabled($storeId) === false) {
-                continue;
-            }
-
-            if (!$this->algoliaCredentialsManager->checkCredentialsWithSearchOnlyAPIKey($storeId)) {
-                $this->algoliaCredentialsManager->displayErrorMessage(self::class, $storeId);
-
-                return;
-            }
-
-            if ($productIds && !$areParentsLoaded) {
-                $productIds = array_unique(array_merge($productIds, $this->productHelper->getParentProductIds($productIds)));
-                $areParentsLoaded = true;
-            }
-
-            $productsPerPage = $this->configHelper->getNumberOfElementByPage();
-
-            if (is_array($productIds) && count($productIds) > 0) {
-                foreach (array_chunk($productIds, $productsPerPage) as $chunk) {
-                    /** @uses ProductIndexBuilder::buildIndexList() */
-                    $this->queue->addToQueue(
-                        ProductIndexBuilder::class,
-                        'buildIndexList',
-                        ['storeId' => $storeId, 'entityIds' => $chunk],
-                        count($chunk)
-                    );
-                }
-
-                continue;
-            }
-
-            $useTmpIndex = $this->configHelper->isQueueActive($storeId);
-            $onlyVisible = !$this->configHelper->includeNonVisibleProductsInIndex();
-            $collection = $this->productHelper->getProductCollectionQuery($storeId, $productIds, $onlyVisible);
-
-            $timerName = __METHOD__ . ' (Get product collection size)';
-            $this->diag->startProfiling($timerName);
-            $size = $collection->getSize();
-            $this->diag->stopProfiling($timerName);
-
-            $pages = ceil($size / $productsPerPage);
-
-            /** @uses IndicesConfigurator::saveConfigurationToAlgolia() */
-            $this->queue->addToQueue(IndicesConfigurator::class, 'saveConfigurationToAlgolia', [
-                'storeId' => $storeId,
-                'useTmpIndex' => $useTmpIndex,
-            ], 1, true);
-            for ($i = 1; $i <= $pages; $i++) {
-                $data = [
-                    'storeId' => $storeId,
-                    'options' => [
-                        'entityIds' => $productIds,
-                        'page' => $i,
-                        'pageSize' => $productsPerPage,
-                        'useTmpIndex' => $useTmpIndex,
-                    ]
-                ];
-
-                /** @uses ProductIndexBuilder::buildIndexFull() */
-                $this->queue->addToQueue(
-                    ProductIndexBuilder::class,
-                    'buildIndexFull',
-                    $data,
-                    $productsPerPage,
-                    true
-                );
-            }
+        if (!$this->configHelper->isProductsIndexerEnabled()) {
```

Comment:
We need to limit our scope on this disable operation to `executeFull` and not `execute` - otherwise this will break how materialized views are handled which is essential for delta updates. (Applies to all affected entity indexers)

Reply:
Makes sense ! Sorry for that.